pkg/export/export.go

```diff
@@ -416,63 +443,96 @@ const (
 )
 
 // ApplyConfig updates the exporter state to the given configuration.
-// Must be called at least once before Export() can be used.
-func (e *Exporter) ApplyConfig(cfg *config.Config) (err error) {
-	// If project_id, location, or cluster were set through the external_labels in the config file,
-	// these values take precedence. If they are unset, the flag value, which defaults to an
-	// environment-specific value on GCE/GKE, is used.
-	builder := labels.NewBuilder(cfg.GlobalConfig.ExternalLabels)
-
-	if !cfg.GlobalConfig.ExternalLabels.Has(KeyProjectID) {
-		builder.Set(KeyProjectID, e.opts.ProjectID)
-	}
-	if !cfg.GlobalConfig.ExternalLabels.Has(KeyLocation) {
-		builder.Set(KeyLocation, e.opts.Location)
+func (e *Exporter) ApplyConfig(cfg *config.Config, opts *ExporterOpts) (err error) {
+	if e.isNopExporter() {
+		return errors.New("nop exporter")
 	}
-	if !cfg.GlobalConfig.ExternalLabels.Has(KeyCluster) {
-		builder.Set(KeyCluster, e.opts.Cluster)
+
+	e.mtx.Lock()
+	defer e.mtx.Unlock()
+
+	optsChanged := false
+	if opts != nil {
+		opts.Default()
+		if err := opts.Validate(); err != nil {
+			return err
+		}
+		optsChanged = !reflect.DeepEqual(e.opts, opts)
+		if optsChanged {
+			e.opts = *opts
+		}
 	}
-	lset := builder.Labels()
+
+	lset := createLabelSet(cfg, &e.opts)
+	labelsChanged := !labels.Equal(e.externalLabels, lset)
```

Comment:
suggestion: Don't check for changed labels

We weren't doing it before. What's the motivation to add it now?